tests/Support/SupportMacroableTest.php

```diff
@@ -42,16 +42,12 @@ public function testWhenCallingMacroClosureIsBoundToObject()
         TestMacroable::macro('tryInstance', function () {
             return $this->protectedVariable;
         });
-        TestMacroable::macro('tryStatic', function () {
-            return static::getProtectedStatic();
-        });
+
         $instance = new TestMacroable;
 
         $result = $instance->tryInstance();
-        $this->assertEquals('instance', $result);
 
-        $result = TestMacroable::tryStatic();
-        $this->assertEquals('static', $result);
```

Comment:
Laravel 6 will lose this feature? Isn't a bit sad to disable the feature (rather than deprecate it) since it's just deprecated and not yet actually removed?

Reply:
If it's deprecated in PHP then it'll be removed eventually. Why wait?

Reply:
By deprecate it PHP warn before removing it. If you remove it now in Laravel there is no warning time for Laravel users to realize there macros no longer work. It's even worst for libraries that would have a range of compatibility like laravel "^5.7 || ^6.0"

Or the opposite case: if you have a Laravel 6.0 app with PHP 7.4, you actually could use `static::` if you did not plan to upgrade yet/soon to PHP 8, Laravel hurry you to refactor your macros.

It's OK to remove in 6.0, but it sounds better to deprecate in a version and remove in the next one. Maybe it could be deprecated in a patch of 5.8.

Reply:
Re-added this.